### PR TITLE
Integration Specs For Views

### PR DESCRIPTION
**Pull request by Mutalenic**
## Project requirements
- Used Capybara to write integration tests for each view in our project.
- Login page:
   - I can see the username and password inputs and the "Submit" button.
   - When a user clicks the submit button without filling in the username and the password, I will get a detailed error.
   - When a user clicks the submit button after filling in the username and the password with incorrect data, I will get a detailed error.
   - When a user clicks the submit button after filling in the username and the password with correct data, I am redirected to the root page.
- User index page:
    -  I can see the username of all other users.
    - I can  see the profile picture for each user.
    - I can see the number of posts each user has written.
    - When a user clicks on a user, a user is redirected to that user's show page.
- user show page:
    - A user can see the user's profile picture.
    - A user can see the user's username.
    - A user can see the number of posts the user has written.
    - I can see the user's bio.
    - I can see the user's first 3 posts.
    - I can see a button that lets me view all of a user's posts.
    - When I click a user's post, it redirects me to that post's show page. 
    - When I click to see all posts, it redirects me to the user's post's index page.
- User post index page:
   - I can see the user's profile picture.
   - I can see the user's username.
   - I can see the number of posts the user has written.
   - I can see a post's title.
   - I can see some of the post's body.
   - I can see the first comments on a post.
   - I can see how many comments a post has.
   - I can see how many likes a post has.
   - I can see a section for pagination if there are more posts than fit on the view.
   - When I click on a post, it redirects me to that post's show page.
- Post show page:
   - I can see the post's title.
   - I can see who wrote the post.
   - I can see how many comments it has.
   - I can see how many likes it has.
   - I can see the post body.
   - I can see the username of each commenter.
   - I can see the comment each commenter left.
To run the test you need to run the following command in your terminal bundle exec rspec spec/views

### CODE REVIEW

File: spec/features/users_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  scenario 'See the number of posts each user has written' do
+    expect(page).to have_content('Number of posts: 0')
+  end
+
+  scenario "When I click on a user, I am redirected to that user's show page" do
+    click_link 'Nicholas', match: :first
+    expect(current_path).to eq user_path(User.first.id)
+  end
+end
```
Comment (medaminedev66) [inline spec/features/users_index_spec.rb]: I think that you have repeated the same test (Testing user index page) twice in this file. Let's eliminate one.

File: spec/features/users_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  scenario "I can see the user's bio" do
+    expect(page).to have_content('Developer')
+  end
+
+  scenario "I can see a button that lets me view all of a user's posts" do
+    click_link('See all posts')
+    expect(current_path).to eq user_posts_path(User.first.id)
+  end
+end
```
Comment (medaminedev66) [inline spec/features/users_show_spec.rb:59]: Please ensure that the User show page has the following featured tests:
- Tests that assert you can see the number of posts the user has written.
- Tests that assert you can see the user's first 3 posts.
- Tests that redirect to the post's show page when you click on a user's post.

File: spec/features/posts_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      expect(page).to have_content 'Likes: 1'
+    end
+    scenario 'I can see the post comments counter.' do
+      expect(page).to have_content 'Comments: 2'
+    end
+    scenario 'I can see the post comments.' do
+      expect(page).to have_content 'My first comment'
+      expect(page).to have_content 'My second comment'
+    end
+  end
```
Comment (medaminedev66) [inline spec/features/posts_index_spec.rb:95]: - [ ] For the Post index page, please ensure that it has the following featured tests:
- Tests that assert you can see the user's profile picture.
- Tests that assert you can see the user's username.
- Tests that assert you can see the number of posts the user has written.
- Tests that redirect to the post's show page when you click on a post.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi team  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- All the integration tests passed.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/users_index_spec.rb (hunk 1 above)
Comment (Mutalenic) [reply spec/features/users_index_spec.rb]: Thanks for the review @medaminedev66 , changes implemented.

File: spec/features/users_show_spec.rb (hunk 2 above)
Comment (Mutalenic) [reply spec/features/users_show_spec.rb:59]: Thanks for the review @medaminedev66 , changes implemented.

File: spec/features/posts_index_spec.rb (hunk 3 above)
Comment (Mutalenic) [reply spec/features/posts_index_spec.rb:95]: Thanks for the review @medaminedev66 , changes implemented.

**Review by descholar-ceo — APPROVED** [review]
Hi @Mutalenic , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**